src/ui/public/registry/field_formats.js

```diff
@@ -1,97 +1,106 @@
 import _ from 'lodash';
-import { uiRegistry } from 'ui/registry/_registry';
+import chrome from 'ui/chrome';
 import { FieldFormat } from '../../field_formats/field_format';
+import { IndexedArray } from 'ui/indexed_array';
 
-export const RegistryFieldFormatsProvider = uiRegistry({
-  name: 'fieldFormats',
-  index: ['id'],
-  group: ['fieldType'],
+class FieldFormatRegistry extends IndexedArray {
+  constructor(config) {
+    const spec = config && config.spec;
+    const iaOpts = _.defaults(_.pick(spec, IndexedArray.OPT_NAMES), { index: ['id'] });
+    super(iaOpts);
 
-  invokeProviders(providers) {
-    // in order to ensure that FieldFormats can be instantiated on the
-    // server and the browser we don't provide them access to the Angular
-    // injector, just the FieldFormat class.
-    return providers.map(createSomeFormat => (
-      createSomeFormat(FieldFormat)
-    ));
-  },
+    this._config = chrome.getUiSettingsClient();
```

Comment:
I know it used to be the `config` service, but now that this class is using the `uiSettings` client can we reflect that in the property naming? `this._uiSettings`?